libheif/CMakeLists.txt

```diff
@@ -68,12 +68,14 @@ endif()
 add_definitions(-DHAVE_VISIBILITY)
 add_definitions(-DLIBHEIF_EXPORTS)
 
-add_library(${LIBHEIF_LIBRARY_NAME} SHARED ${libheif_sources})
+add_library(${PROJECT_NAME} SHARED ${libheif_sources})
```

Comment:
I think this is wrong, as the PROJECT_NAME is libheif which would result in liblibheif. In the meantime you do:

```
add_library(heif ${libheif_sources})
target_link_libraries(heif ...)

add_library(heif::heif ALIAS heif)
```
If you want to link against heif, use the alias `heif::heif`. This way you can also modify it.

instead of add_library(... SHARED ...) you remove the shared and add:

```
option(BUILD_SHARED_LIBS "Build shared libraries" ON)
```

Then you can easily build as a static lib too. This is what the documentation for BUILD_SHARED_LIBS suggest.

Reply:
Did you read the comment I posted here?
https://github.com/strukturag/libheif/pull/168/files#r337861735

The original name of the project was `libheif` and I wanted to change as little as possible to avoid having a bigger impact in codebases consuming **libheif**. I also think it makes sense to use `${PROJECT_NAME}`  as the library target name to avoid hardcoded values. 

I think I did not understand you completely about your point with the line
```
add_library(heif::heif ALIAS heif)
```
In case we would want to add a namespace for the library target, should not we use the NAMESPACE option in the `install(EXPORT ...)` command?
https://cmake.org/cmake/help/v3.13/command/install.html#installing-exports

Finally, I completely agree with the `BUILD_SHARED_LIBS` suggestion. It is also something I normally use in my projects. I will add it.

Reply:
I just pushed to additional commit adding the support for `BUILD_SHARED_LIBS` and other one changing the name of the project to `heif` (more information in the commit message).